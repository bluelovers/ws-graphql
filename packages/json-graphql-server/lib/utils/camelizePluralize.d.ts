export default function camelizePluralize(name: string): string;
