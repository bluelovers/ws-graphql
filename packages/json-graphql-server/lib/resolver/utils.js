"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.findEntityIndex = void 0;
function findEntityIndex(id, entityData) {
    const stringId = id.toString();
    const indexOfEntity = entityData.findIndex(e => e.id != null && e.id.toString() === stringId);
    return indexOfEntity;
}
exports.findEntityIndex = findEntityIndex;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoidXRpbHMuanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyJ1dGlscy50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOzs7QUFFQSxTQUFnQixlQUFlLENBQTRELEVBQVUsRUFBRSxVQUFlO0lBRXJILE1BQU0sUUFBUSxHQUFHLEVBQUUsQ0FBQyxRQUFRLEVBQUUsQ0FBQztJQUMvQixNQUFNLGFBQWEsR0FBRyxVQUFVLENBQUMsU0FBUyxDQUN6QyxDQUFDLENBQUMsRUFBRSxDQUFDLENBQUMsQ0FBQyxFQUFFLElBQUksSUFBSSxJQUFJLENBQUMsQ0FBQyxFQUFFLENBQUMsUUFBUSxFQUFFLEtBQUssUUFBUSxDQUNqRCxDQUFDO0lBRUYsT0FBTyxhQUFhLENBQUE7QUFDckIsQ0FBQztBQVJELDBDQVFDIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IHsgSVNvdXJjZURhdGFSb3dCYXNlQ29yZSB9IGZyb20gJy4uL3R5cGVzJztcblxuZXhwb3J0IGZ1bmN0aW9uIGZpbmRFbnRpdHlJbmRleDxUIGV4dGVuZHMgSVNvdXJjZURhdGFSb3dCYXNlQ29yZSA9IElTb3VyY2VEYXRhUm93QmFzZUNvcmU+KGlkOiBzdHJpbmcsIGVudGl0eURhdGE6IFRbXSlcbntcblx0Y29uc3Qgc3RyaW5nSWQgPSBpZC50b1N0cmluZygpO1xuXHRjb25zdCBpbmRleE9mRW50aXR5ID0gZW50aXR5RGF0YS5maW5kSW5kZXgoXG5cdFx0ZSA9PiBlLmlkICE9IG51bGwgJiYgZS5pZC50b1N0cmluZygpID09PSBzdHJpbmdJZCxcblx0KTtcblxuXHRyZXR1cm4gaW5kZXhPZkVudGl0eVxufVxuIl19