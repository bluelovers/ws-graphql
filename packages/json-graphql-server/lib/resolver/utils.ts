import { ISourceDataRowBaseCore } from '../types';

export function findEntityIndex<T extends ISourceDataRowBaseCore = ISourceDataRowBaseCore>(id: string, entityData: T[])
{
	const stringId = id.toString();
	const indexOfEntity = entityData.findIndex(
		e => e.id != null && e.id.toString() === stringId,
	);

	return indexOfEntity
}
