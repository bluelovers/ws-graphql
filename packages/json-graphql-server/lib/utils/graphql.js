"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.printSchemaForTypes = exports.printSchemaForFields = void 0;
const graphql_1 = require("graphql");
/**
 * Return a schema string with a Main type using the fields
 *
 * @param {*array} fields
 *
 * @example
 * printSchemaForFields({
 *     id: { type: graphql.GraphQLString },
 *     title: { type: graphql.GraphQLString },
 *     views: { type: graphql.GraphQLInt },
 *     user_id: { type: graphql.GraphQLString },
 * });
 * // type Main {
 * //   id: String
 * //   title: String
 * //   views: String
 * //   user_id: String
 * // }
 * //
 * // type Query {
 * //   foo: Main
 * // }
 */
function printSchemaForFields(fields) {
    const mainType = new graphql_1.GraphQLObjectType({
        name: 'Main',
        fields,
    });
    const queryType = new graphql_1.GraphQLObjectType({
        name: 'Query',
        fields: {
            foo: { type: mainType },
        },
    });
    const schema = new graphql_1.GraphQLSchema({ query: queryType });
    return graphql_1.printSchema(schema);
}
exports.printSchemaForFields = printSchemaForFields;
function printSchemaForTypes(types) {
    const typesSchema = types.reduce((schema, type) => {
        schema[type.name] = type;
        return schema;
    }, {});
    const queryType = new graphql_1.GraphQLObjectType({
        name: 'Query',
        fields: types.reduce((fields, type) => {
            fields[type.name] = { type };
            return fields;
        }, {}),
    });
    const schema = new graphql_1.GraphQLSchema({ ...typesSchema, query: queryType });
    return graphql_1.printSchema(schema);
}
exports.printSchemaForTypes = printSchemaForTypes;
//# sourceMappingURL=graphql.js.map