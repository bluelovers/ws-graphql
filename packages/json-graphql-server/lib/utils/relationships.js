"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isRelationshipField = void 0;
function isRelationshipField(fieldName) {
    return fieldName !== '_id' && fieldName.endsWith('_id');
}
exports.isRelationshipField = isRelationshipField;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoicmVsYXRpb25zaGlwcy5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbInJlbGF0aW9uc2hpcHMudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7O0FBQUEsU0FBZ0IsbUJBQW1CLENBQUMsU0FBaUI7SUFFcEQsT0FBTyxTQUFTLEtBQUssS0FBSyxJQUFJLFNBQVMsQ0FBQyxRQUFRLENBQUMsS0FBSyxDQUFDLENBQUM7QUFDekQsQ0FBQztBQUhELGtEQUdDIiwic291cmNlc0NvbnRlbnQiOlsiZXhwb3J0IGZ1bmN0aW9uIGlzUmVsYXRpb25zaGlwRmllbGQoZmllbGROYW1lOiBzdHJpbmcpXG57XG5cdHJldHVybiBmaWVsZE5hbWUgIT09ICdfaWQnICYmIGZpZWxkTmFtZS5lbmRzV2l0aCgnX2lkJyk7XG59XG4iXX0=