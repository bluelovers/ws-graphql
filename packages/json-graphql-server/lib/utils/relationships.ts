export function isRelationshipField(fieldName: string)
{
	return fieldName !== '_id' && fieldName.endsWith('_id');
}
