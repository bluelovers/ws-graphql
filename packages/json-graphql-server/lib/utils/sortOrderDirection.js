"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnumOrderDirection = void 0;
var EnumOrderDirection;
(function (EnumOrderDirection) {
    EnumOrderDirection[EnumOrderDirection["ASC"] = 1] = "ASC";
    EnumOrderDirection[EnumOrderDirection["DESC"] = -1] = "DESC";
})(EnumOrderDirection = exports.EnumOrderDirection || (exports.EnumOrderDirection = {}));
function sortOrderDirection(sortOrder) {
    if (sortOrder === true || (typeof sortOrder === 'number' && sortOrder > 0) || String(sortOrder).toLowerCase() == 'asc') {
        return 1 /* ASC */;
    }
    return -1 /* DESC */;
}
exports.default = sortOrderDirection;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoic29ydE9yZGVyRGlyZWN0aW9uLmpzIiwic291cmNlUm9vdCI6IiIsInNvdXJjZXMiOlsic29ydE9yZGVyRGlyZWN0aW9uLnRzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiI7OztBQUVBLElBQWtCLGtCQUlqQjtBQUpELFdBQWtCLGtCQUFrQjtJQUVuQyx5REFBTyxDQUFBO0lBQ1AsNERBQVMsQ0FBQTtBQUNWLENBQUMsRUFKaUIsa0JBQWtCLEdBQWxCLDBCQUFrQixLQUFsQiwwQkFBa0IsUUFJbkM7QUFFRCxTQUF3QixrQkFBa0IsQ0FBQyxTQUFzRTtJQUVoSCxJQUFJLFNBQVMsS0FBSyxJQUFJLElBQUksQ0FBQyxPQUFPLFNBQVMsS0FBSyxRQUFRLElBQUksU0FBUyxHQUFHLENBQUMsQ0FBQyxJQUFJLE1BQU0sQ0FBQyxTQUFTLENBQUMsQ0FBQyxXQUFXLEVBQWdCLElBQUksS0FBSyxFQUNwSTtRQUNDLG1CQUE2QjtLQUM3QjtJQUVELHFCQUE4QjtBQUMvQixDQUFDO0FBUkQscUNBUUMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgeyBJU29ydE9yZGVyIH0gZnJvbSAnLi4vdHlwZXMnO1xuXG5leHBvcnQgY29uc3QgZW51bSBFbnVtT3JkZXJEaXJlY3Rpb25cbntcblx0QVNDID0gMSxcblx0REVTQyA9IC0xLFxufVxuXG5leHBvcnQgZGVmYXVsdCBmdW5jdGlvbiBzb3J0T3JkZXJEaXJlY3Rpb24oc29ydE9yZGVyOiBJU29ydE9yZGVyIHwgc3RyaW5nIHwgYm9vbGVhbiB8IEVudW1PcmRlckRpcmVjdGlvbiB8IDEgfCAtMSk6IEVudW1PcmRlckRpcmVjdGlvbi5BU0MgfCBFbnVtT3JkZXJEaXJlY3Rpb24uREVTQ1xue1xuXHRpZiAoc29ydE9yZGVyID09PSB0cnVlIHx8ICh0eXBlb2Ygc29ydE9yZGVyID09PSAnbnVtYmVyJyAmJiBzb3J0T3JkZXIgPiAwKSB8fCBTdHJpbmcoc29ydE9yZGVyKS50b0xvd2VyQ2FzZSgpIGFzIElTb3J0T3JkZXIgPT0gJ2FzYycpXG5cdHtcblx0XHRyZXR1cm4gRW51bU9yZGVyRGlyZWN0aW9uLkFTQ1xuXHR9XG5cblx0cmV0dXJuIEVudW1PcmRlckRpcmVjdGlvbi5ERVNDXG59XG4iXX0=