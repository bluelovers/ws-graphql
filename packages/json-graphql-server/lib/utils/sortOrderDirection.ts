import { ISortOrder } from '../types';

export const enum EnumOrderDirection
{
	ASC = 1,
	DESC = -1,
}

export default function sortOrderDirection(sortOrder: ISortOrder | string | boolean | EnumOrderDirection | 1 | -1): EnumOrderDirection.ASC | EnumOrderDirection.DESC
{
	if (sortOrder === true || (typeof sortOrder === 'number' && sortOrder > 0) || String(sortOrder).toLowerCase() as ISortOrder == 'asc')
	{
		return EnumOrderDirection.ASC
	}

	return EnumOrderDirection.DESC
}
