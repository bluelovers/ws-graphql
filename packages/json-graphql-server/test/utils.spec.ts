/**
 * Created by user on 2020/3/18.
 */

import {
	getRelatedTypeUnsafe,
	getRelationshipFromKey,
	getTypeFromKey,
	camelizePluralize,
	getRelatedKey,
	getReverseRelatedField, getRelatedKeyFromType, getRelatedTypeWithValid,
} from '../lib/utils/nameConverter';
import { GraphQLObjectType } from 'graphql';
import sortOrderDirection, { EnumOrderDirection } from '../lib/utils/sortOrderDirection';
import { isRelationshipField } from '../lib/utils/relationships';

describe(`nameConverter`, () => {

	test(`getRelationshipFromKey('users')`, () => {

		let actual = getRelationshipFromKey('users');
		let expected = 'Users';

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date)
		expect(actual).toMatchSnapshot();

	});

	test(`getTypeFromKey('users')`, () => {

		let actual = getTypeFromKey('users');
		let expected = 'User';

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date)
		expect(actual).toMatchSnapshot();

	});

	test(`camelizePluralize('user')`, () => {

		let actual = camelizePluralize('user');
		let expected = 'Users';

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date)
		expect(actual).toMatchSnapshot();

	});

	test(`getRelatedKey('user_id')`, () => {

		let actual = getRelatedKey('user_id');
		let expected = 'users';

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date)
		expect(actual).toMatchSnapshot();

	});

	test(`getReverseRelatedField('users')`, () => {

		let actual = getReverseRelatedField('users');
		let expected = 'user_id';

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date)
		expect(actual).toMatchSnapshot();

	});

	test(`getRelatedTypeUnsafe('user_id')`, () => {

		let actual = getRelatedTypeUnsafe('user_id');
		let expected = 'User';

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date)
		expect(actual).toMatchSnapshot();

	});

	test(`getRelatedTypeWithValid`, () => {

		let actual = getRelatedTypeWithValid('user_id');
		let expected = 'User';

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date)
		expect(actual).toMatchSnapshot();

		actual = getRelatedTypeWithValid('_id');
		expect(actual).toStrictEqual('');
		expect(actual).toMatchSnapshot();

	});

	test(`getRelatedKeyFromType(User)`, () => {

		let actual = getRelatedKeyFromType({
			toString()
			{
				return 'User'
			}
		} as any as GraphQLObjectType);
		let expected = 'Users';

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date)
		expect(actual).toMatchSnapshot();

	});

})

describe(`utils`, () => {

	test(`isRelationshipField`, () =>
	{

		let actual = isRelationshipField('user_id');
		let expected = true;

		expect(actual).toStrictEqual(expected);
		expect(actual).toMatchSnapshot();

		actual = isRelationshipField('_id');
		expected = false;

		expect(actual).toStrictEqual(expected);
		expect(actual).toMatchSnapshot();
	});

	test(`sortOrderDirection`, () => {

		let actual: EnumOrderDirection;
		let expected: EnumOrderDirection;

		expected = EnumOrderDirection.ASC;

		actual = sortOrderDirection('asc');
		expect(actual).toStrictEqual(expected);
		expect(actual).toMatchSnapshot();

		actual = sortOrderDirection('ASC');
		expect(actual).toStrictEqual(expected);
		expect(actual).toMatchSnapshot();

		actual = sortOrderDirection(true);
		expect(actual).toStrictEqual(expected);
		expect(actual).toMatchSnapshot();

		expected = EnumOrderDirection.DESC;

		actual = sortOrderDirection('desc');
		expect(actual).toStrictEqual(expected);
		expect(actual).toMatchSnapshot();

		actual = sortOrderDirection('DESC');
		expect(actual).toStrictEqual(expected);
		expect(actual).toMatchSnapshot();

		actual = sortOrderDirection(false);
		expect(actual).toStrictEqual(expected);
		expect(actual).toMatchSnapshot();

		actual = sortOrderDirection(null);
		expect(actual).toStrictEqual(expected);
		expect(actual).toMatchSnapshot();

		actual = sortOrderDirection(void 0);
		expect(actual).toStrictEqual(expected);
		expect(actual).toMatchSnapshot();
	});

})
