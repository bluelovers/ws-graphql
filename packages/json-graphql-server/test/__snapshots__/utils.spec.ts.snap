// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nameConverter camelizePluralize('user') 1`] = `"Users"`;

exports[`nameConverter getRelatedKey('user_id') 1`] = `"users"`;

exports[`nameConverter getRelatedKeyFromType(User) 1`] = `"Users"`;

exports[`nameConverter getRelatedTypeUnsafe('user_id') 1`] = `"User"`;

exports[`nameConverter getRelatedTypeWithValid 1`] = `"User"`;

exports[`nameConverter getRelatedTypeWithValid 2`] = `""`;

exports[`nameConverter getRelationshipFromKey('users') 1`] = `"Users"`;

exports[`nameConverter getReverseRelatedField('users') 1`] = `"user_id"`;

exports[`nameConverter getTypeFromKey('users') 1`] = `"User"`;

exports[`utils isRelationshipField 1`] = `true`;

exports[`utils isRelationshipField 2`] = `false`;

exports[`utils sortOrderDirection 1`] = `1`;

exports[`utils sortOrderDirection 2`] = `1`;

exports[`utils sortOrderDirection 3`] = `1`;

exports[`utils sortOrderDirection 4`] = `-1`;

exports[`utils sortOrderDirection 5`] = `-1`;

exports[`utils sortOrderDirection 6`] = `-1`;

exports[`utils sortOrderDirection 7`] = `-1`;

exports[`utils sortOrderDirection 8`] = `-1`;
